File: APIMASH_CNorris_StarterKit/APIMASH_CNorrisLib/APIMASH_OM_Bindable.cs
using System;
using System.ComponentModel;
using System.Runtime.CompilerServices;

/*
* LICENSE: https://raw.github.com/apimash/StarterKits/master/LicenseTerms-SampleApps%20.txt
*/

namespace APIMASH_CNorrisLib
{
    [Windows.Foundation.Metadata.WebHostHidden]
    public class APIMASH_OM_Bindable : INotifyPropertyChanged
    {
        public string Joke { get; set; }

        public event PropertyChangedEventHandler PropertyChanged;

        protected bool SetProperty<T>(ref T storage, T value, [CallerMemberName] String propertyName = null)
        {
            if (object.Equals(storage, value)) return false;

            storage = value;
            this.OnPropertyChanged(propertyName);
            return true;
        }

        protected void OnPropertyChanged([CallerMemberName] string propertyName = null)
        {
            var eventHandler = this.PropertyChanged;
            if (eventHandler != null)
            {
                eventHandler(this, new PropertyChangedEventArgs(propertyName));
            }
        }
    }
}

File: Windows Starter Kits/APIMASH_CNorris_VB_StarterKit/APIMASH_CNorrisVBLib/APIMASH_OM_Bindable.vb
Imports System
Imports System.ComponentModel
Imports System.Runtime.CompilerServices
'
'* LICENSE: https://raw.github.com/apimash/StarterKits/master/LicenseTerms-SampleApps%20.txt
'
Namespace APIMASH_CNorrisLib
    <Windows.Foundation.Metadata.WebHostHidden> _
    Public Class APIMASH_OM_Bindable
        Implements INotifyPropertyChanged

        Public Property Joke() As String
            Get
                Return m_Joke
            End Get
            Set(value As String)
                m_Joke = value
            End Set
        End Property
        Private m_Joke As String

        Public Event PropertyChanged As PropertyChangedEventHandler

        Protected Function SetProperty(Of T)(ByRef storage As T, value As T, <CallerMemberName> Optional propertyName As [String] = Nothing) As Boolean
            If Object.Equals(storage, value) Then
                Return False
            End If

            storage = value
            Me.OnPropertyChanged(propertyName)
            Return True
        End Function

        Protected Sub OnPropertyChanged(<CallerMemberName> Optional propertyName As String = Nothing)
            'Dim eventHandler = Me.PropertyChanged
            RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
        End Sub

        Public Event PropertyChanged1(sender As Object, e As PropertyChangedEventArgs) Implements INotifyPropertyChanged.PropertyChanged
    End Class
End Namespace


